bcl2fastq/index_summary.py

```diff
@@ -5,11 +5,14 @@
 import dash.dependencies as dep
 import pandas
 
-
 index = gsiqcetl.bcl2fastq.parse.load_cache(
     gsiqcetl.bcl2fastq.parse.CACHENAME.SAMPLES,
     './data/bcl2fastq_cache.hd5'
 )
+# If both Index1 and Index2 are NaN, this will filter out those indexes. This is to prevent run
+# 180810_M00753_0053_000000000-D4HKJ from crashing 
+if type(index[index['Index1'].notna()]) == float and type(index[index['Index2'].nota()]) == float:
```

Comment:
What is the purpose of this line? `type(index[index['Index1'].notna()]) == float` will always be False, as `type(index[index['Index1'].notna()])` is `DataFrame`.

Reply:
My mistake, this should be .isna(). If I make this change, would that work?

Reply:
I don't believe so. What is the purpose of this if statement?

Reply:
This was to filter by NaN, but I think I read the description wrong. NaN doesn't have type float so this statement wouldn't execute as I wanted.

Reply:
NaN is of type float

![image](https://user-images.githubusercontent.com/37381890/59777883-71418080-9283-11e9-85fc-b00b5de9d1dc.png)

The line below (`index = index[~index["Index1"].isna()]`) does the filtering, so I am still not sure what the `if` statement aims to do.

Reply:
It would only execute the filtering if both Index1 and Index 2 were float

Reply:
I understand. The `if` statement is not necessary. Once you remove 

https://github.com/ijafri26/dashi/blob/b73ad578089106ab89ecfccafaefc3282c71f824/bcl2fastq/index_summary.py#L99

you will see that the bug persists with your current code.

I think there is a misunderstanding in how DataFrame filtering works. I would suggest reading https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html and https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html